Frontend/src/pages/ExploreSystems.js

```diff
@@ -486,10 +486,38 @@ function ExploreSystems() {
                                     >
                                         Rational Preperiodic Points
                                     </span>
-                                    <ul className="nested"></ul>
+                                    <ul className="nested">
+                                        <li>
+                                            <input
+                                                type="number"
+                                                style={textBoxStyle}
+                                                value={filters.rationalPreperiodicCardinality || ''}
+                                                onChange={(e) => handleTextChange("rationalPreperiodicCardinality", e.target.value)}
+                                            />
+                                            <label>Cardinality</label>
+                                        </li>
+                                        <li>
+                                            <input
+                                                type="number"
+                                                style={textBoxStyle}
+                                                value={filters.rationalPreperiodicComponents || ''}
+                                                onChange={(e) => handleTextChange("rationalPreperiodicComponents", e.target.value)}
+                                            />
+                                            <label>Number of Connected Components</label>
```

Comment:
While the name is requested, it is a little long and it causes the UI to look a little cluttered. Consider reducing the length of the name so that it doesn't automatically wrap

Reply:
fixed by shortening "number of connected components" to "no of connected components" for now